### PR TITLE
Sync fork

### PR DESCRIPTION
**Pull request by peterdeme**
- Rebased to the main branch of upstream
- Made sure all of Kuba's updates are still included
- Upstream had a bunch of tests that are difficult to run so I replaced them with `.github/workflows/build.yml` simply

### CODE REVIEW

**Comment by peterdeme** [comment]
@cube2222 @michieldewilde - the primary reason of this PR is that there are underlying dependencies that had breaking changes (specifically pprof). this makes it impossible for us to bump packages in our `backend` repository. a sync solves this issue.

image

Link for the run: https://github.com/spacelift-io/backend/actions/runs/9810725319/job/27091488683?pr=7365

File: go.mod [hunk 1]
```diff
@@ -1,221 +1,283 @@
...
-	github.com/gorilla/mux v1.7.1
-	github.com/graph-gophers/graphql-go v1.3.0
-	github.com/hashicorp/consul/api v1.0.0
-	github.com/hashicorp/vault/api v1.1.0
-	github.com/hashicorp/vault/sdk v0.1.14-0.20200519221838-e0cfd64bc267
-	github.com/jackc/pgx/v4 v4.14.0
-	github.com/jinzhu/gorm v1.9.10
-	github.com/jmoiron/sqlx v1.2.0
-	github.com/julienschmidt/httprouter v1.2.0
+	github.com/google/pprof v0.0.0-20230817174616-7a8ec2ada47b
```
Comment (peterdeme) [inline go.mod:55]: this bump solves our issues :)

**Comment by peterdeme** [comment]
@cube2222 - can you make the `main` branch is the main branch of this repo? currently it's `v1`

**Comment by cube2222** [comment]
@peterdeme done

**Review by eliecharra — APPROVED** [review]
I did not reviewed the ~600 changed files, I trust you on this one 😅

**Pull request merged**